### DISCUSSION TITLE
What is loss/epoch?

### DISCUSSION THREAD

**Discussion by Seedmanc** [67372a/LoRA_Easy_Training_Scripts#28] [Q&A]
It looks neither like validation loss, nor like the old regular loss.
![image](https://github.com/user-attachments/assets/28fe5c1a-1a54-49cb-9d3c-4c39ef06adc3)
Here it suggests overfitting at the end, but val loss doesn't really go up as much, and the old loss is just random (as it always was).

**Comment by 67372a** [comment]
@Seedmanc Loss per epoch is likely not accurate or precise.
logs = {"loss/epoch": loss_recorder.moving_average}
So it would be better to just smooth current_loss and use that for a more accurate state of the training loss.

**Reply by Seedmanc** [reply]
But can you tell how it's calculated? It's clearly different from other losses, and it's the only loss that I get NaNs in sometimes. Asking because it's also only curve that has the characteristic U-shape. I can't judge overfitting by val loss, it never goes up for me.

**Reply by 67372a** [reply]
@Seedmanc This is how, https://github.com/67372a/sd-scripts/blob/5d567ed3c0d047a65f8412f4965e36a1ffb73c47/library/train_util.py#L7572
